core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -351,6 +416,29 @@ private void tryStart() {
         }
     }
 
+    @SneakyThrows(JsonProcessingException.class)
+    final String hash(CreateContainerCmd createCommand) {
+        // TODO add Testcontainers' version to the hash
+        String commandJson = new ObjectMapper()
+            .enable(MapperFeature.SORT_PROPERTIES_ALPHABETICALLY)
+            .enable(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS)
+            .writeValueAsString(createCommand);
+
+        return DigestUtils.sha1Hex(commandJson);
+    }
+
+    Optional<String> findContainerForReuse(String hash) {
+        // TODO locking
+        return dockerClient.listContainersCmd()
+            .withLabelFilter(ImmutableMap.of(HASH_LABEL, hash))
+            .withLimit(1)
+            .withStatusFilter(Arrays.asList("running"))
```

Comment:
```suggestion
            .withStatusFilter(Collections.singletonList("running"))
```

Reply:
any specific reason why? 
1. `asList` can be extended with more options, `singletonList` will require a change. 
2. `asList` is simply shorter
3. immutability here is unnecessary since we're not passing the object outside

looks like 2:0 to me :)

Reply:
No better reason but immutability. 1) is a good argument.